src/emmy/differential.cljc

```diff
@@ -337,9 +338,35 @@
     modified by replacing any appearance of `old-tag` with `new-tag`. Else,
     return `this`.")
 
-  (extract-tangent [this tag]
+  (extract-tangent [this tag mode]
     "If `this` is perturbed, return the tangent component paired with the
-    supplied tag. Else, returns `([[emmy.value/zero-like]] this)`."))
+    supplied tag. Else, returns `([[emmy.value/zero-like]] this)`.")
+
+  (extract-id [this id]))
+
+(defrecord Completed [v->partial]
```

Comment:
annoying that this now has to live in `differential`. It has to be here so that a `Dual` can return the correct default type if it's called with a different tag.